meteor/sofie-server/data-access/repositories/cache/cached-rundown-repository.ts

```diff
@@ -0,0 +1,48 @@
+import { RundownRepository } from '../interfaces/rundown-repository'
+import { Rundown } from '../../../model/entities/rundown'
+import { MisconfigurationException } from '../../../model/exceptions/misconfiguration-exception'
+import { Identifier } from '../../../model/interfaces/identifier'
+
+export class CachedRundownRepository implements RundownRepository {
+	private static instance: RundownRepository
+
+	public static getInstance(rundownRepository?: RundownRepository): RundownRepository {
+		if (!this.instance) {
+			if (!rundownRepository) {
+				throw new MisconfigurationException(
+					`No RundownRepository provided. Unable to create instance of ${CachedRundownRepository.name}`
+				)
+			}
+			this.instance = new CachedRundownRepository(rundownRepository)
+		}
+		return this.instance
+	}
+
+	private rundownRepository: RundownRepository
+
+	private cachedRundowns: Map<string, Rundown> = new Map()
+
+	constructor(rundownRepository: RundownRepository) {
+		this.rundownRepository = rundownRepository
+	}
+
+	public async getRundown(rundownId: string): Promise<Rundown> {
+		if (!this.cachedRundowns.has(rundownId)) {
+			console.log(`### Rundown with id: "${rundownId}" not found in cache. Loading rundown from database...`)
+			const rundown: Rundown = await this.rundownRepository.getRundown(rundownId)
+			this.cachedRundowns.set(rundownId, rundown)
+		}
+		return this.cachedRundowns.get(rundownId) as Rundown
```

Comment:
We could return a clone if that's preferred. I currently don't have any arguments against it.

If we implement that it returns a clone of the Rundown then it will be treated as "transactions" and discarded by default if something throws.

This works because we call our repository (in this case the cache) everytime we want to do something with our Rundown.
If something throws before we manages to call save. Then our execution is aborted, nothing is saved and the next time we call the execution we call the repository again to get the Rundown and the Rundown we get is the one from before we started the flow that got aborted i.e. all the previous "transactions" has been "discarded".